lib/ar-find-in-batches-with-order.rb

```diff
@@ -53,7 +56,9 @@ def find_each_with_order(options = {})
       end
     end
 
-
+    def get_property_val(record, parent_key, property_key)
+      parent_key ? record.as_json(include: parent_key).dig(parent_key, property_key) : record.send(property_key)
```

Comment:
now that I think about it, this is probably kind of expensive to call, and below would be more efficient
`parent_key ? record.send(parent_key).send(property_key) : record.send(property_key)`